state/executor.go

```diff
@@ -290,42 +319,120 @@ func (e *BlockExecutor) execute(ctx context.Context, state types.State, block *t
 		}
 	})
 
+	genAndGossipFraudProofIfNeeded := func(beginBlockRequest *abci.RequestBeginBlock, deliverTxRequests []*abci.RequestDeliverTx, endBlockRequest *abci.RequestEndBlock) (err error) {
+		if fraudProofsEnabled {
```

Comment:
nit: I think it is cleaner in invert these conditions to keep the code more inline.

```go
if !fraudProofsEnabled {
  return nil
}
...
```